src/cudnn/conv.jl

```diff
@@ -10,9 +10,40 @@ using CUDA.CUDNN: scalingParameter, CUDNN_CONVOLUTION, convdims,
 
 const CUDNNFloat = Union{Float16,Float32,Float64}
 
-function cudnnConvolutionDescriptor(cdims::DenseConvDims, x::DenseCuArray{T}) where T
+function cudnnConvolutionDescriptorAndPaddedInput(cdims::DenseConvDims, x::DenseCuArray{T}) where T
+    # The main purpose of this function is to catch asymmetric padding which cudnn does not support
+    # If we find asymmetric padding we'll make a copy of x which is manually padded so that we can
+    # call cudnn with symmetric padding.
+    pad = collect(NNlib.padding(cdims)) # work with an array to make things more type stable
+    all(pad[1:2:end] .== pad[2:2:end]) && return (cudnnConvolutionDescriptor(cdims, x), x, identity)
+
+    # Maybe we should warn the user that this copies data, but other ML libs generally don't warn
```

Comment:
A note in the main NNlib + Flux docs should be enough IMO.

Reply:
Cool, removed the comment. I will look around for a place where it fits but feel free to add it yourself if you know the place.